dart/lib/src/noop_sentry_span.dart

```diff
@@ -26,7 +26,7 @@ class NoOpSentrySpan extends ISentrySpan {
   }
 
   @override
-  Future<void> finish({SpanStatus? status}) async {}
+  Future<void> finish({SpanStatus? status, DateTime? endTimestamp}) async {}
```

Comment:
Wondering if it makes sense to be `finishTimestamp`, the method is called `finish` and not `end`.
Opinions @brustolin ?

Reply:
Well, the property being manipulated by this parameter is called `timestamp`, maybe only timestamp would be enough. 

But I don't have strong opinions about parameter naming. Either option sounds good to me. The purpose of the parameter is clear (at least for me, it might be because I'm used to the SDK).

Reply:
Since there is now one more time stamp, `startTimestamp`, I renamed timestamp as `endTimestamp` in this PR. Just like PHP and JS repo did.
And endTimestamp is already in use in the [same class](https://github.com/getsentry/sentry-dart/blob/7a39898ffa64fd4b00b0e7e5f0337ce9e22ab382/dart/lib/src/protocol/sentry_span.dart#L119):
```dart
#118  @override
#119  DateTime? get endTimestamp => _timestamp;
```

I can revert it to timestamp and rename the method parameter to the same name if it make sense.

Reply:
`endTimestamp` is the correct one, it's also part of our Unified API https://develop.sentry.dev/sdk/performance/
The serialization name should not change tho, since it's part of our protocol during event ingestion.

Reply:
`endTimestamp` is the current one, and the serialization name is left [as is](https://github.com/getsentry/sentry-dart/blob/bce00c31543b75ba9b017a6bad5df4d784af9ead/dart/lib/src/protocol/sentry_span.dart#L151).